libyul/optimiser/FunctionCallFinder.h

```diff
@@ -0,0 +1,42 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/**
+ * AST walker that finds all calls to a function of a given name.
+ */
+
+#pragma once
+
+#include <libyul/optimiser/ASTWalker.h>
+
+#include <vector>
+
+namespace solidity::yul
+{
+
+class FunctionCallFinder: ASTModifier
```

Comment:
Please document.

This class might become handy for the function inliner - currently we check functions to inline to, but maybe we should rather check functions to inline everywhere.

Reply:
Why is it a Modifier?

Reply:
Please also document that this requires unique names.

Reply:
I need non-const pointers (I'll rewrite the argument to ``memoryinit`` using this)... I *could* simultanously implement a const/ASTWalker and a non-const/ASTModifier version with some template magic... otherwise we'd need two versions of this, if we want a const version...

Reply:
Ah, then it makes sense!